### ISSUE TITLE
Narrowing interface fields in an implementation fails

### ISSUE THREAD

**Issue by lirbank** [neo4j/graphql#213]
Consider the following example:

```ts
import { Neo4jGraphQL } from '@neo4j/graphql';

new Neo4jGraphQL({
  typeDefs: /* GraphQL */ `
    interface Vehicle {
      id: ID!
      color: String # NOTE: 'color' is optional on the interface
    }

    type Car implements Vehicle {
      id: ID!
      color: String! # NOTE: 'color' is mandatory on the type, which should be okay
    }

    type Query {
      cars: [Vehicle!]!
    }
  `,
});
```
Here, `color` is defined as an optional string in the Vehicle interface. In the Car implementation, the `color` field is mandatory. In GraphQL an optional type in an interface may be implemented as mandatory. Trying to do this with Neo4jGraphQL fails with:

```
Error: type Car does not implement interface Vehicle correctly
```

**System:**

"@neo4j/graphql": "1.0.1",

**Comment by danstarns** [comment]
Hi. The error you are seeing is correct, `Vehicle.color` is of type String whereas `Car.color` is of type String NonNull. Change `Car.color` to non-null and your issue should go.

**Comment by darrellwarde** [comment]
I initially thought @danstarns was on point here, but after experimenting it does seem valid to do what you have described here. It's on our books to look at schema validation, so we will look into this particular case whilst we do that work. 🙂

**Comment by danstarns** [comment]
Copverage in PR #200

**Comment by darrellwarde** [comment]
This has been fixed by #300, which will be released in version 2.0.0.

**Comment by darrellwarde** [comment]
In an effort to clean our issue backlog a bit, going to close this one off as the fix is available in any 2.0.0 release whenever you decide to upgrade, be it during prerelease or after stable.
